docs/src/main/asciidoc/amazon-lambda.adoc

```diff
@@ -240,33 +294,469 @@ of things you need to do.  Take a look at the generated example project to get a
+Run the following SAM CLI command to locally test your lambda function, passing the appropriate SAM `template`.
+The `event` parameter takes any JSON file, in this case the sample `payload.json`.
+
+NOTE: If using Gradle, the path to the binaries in the YAML templates must be changed from `target` to `build`
 
 [source]
 ----
-sam local invoke --template sam.jvm.yaml --event {json test file}
+sam local invoke --template sam.jvm.yaml --event payload.json
 ----
 
 The native image can also be locally tested using the `sam.native.yaml` template:
 
 [source]
 ----
-sam local invoke --template sam.native.yaml --event {json test file}
+sam local invoke --template sam.native.yaml --event payload.json
 ----
 
 == Tracing with AWS XRay and GraalVM
 
 If you are building native images, and want to use https://aws.amazon.com/xray[AWS X-Ray Tracing] with your lambda
 you will need to include `quarkus-amazon-lambda-xray` as a dependency in your pom.  The AWS X-Ray
 library is not fully compatible with GraalVM so we had to do some integration work to make this work.
+
+In addition, remember to enable the AWS X-Ray tracing parameter in manage.sh, in the cmd_create() function.  This can also be set in the AWS Management Console.
+[source]
+----
+    --tracing-config Mode=Active
+----
+
+For the sam template files, add the following to the YAML function Properties.
+[source]
+----
+    Tracing: Active
+----
+
+AWS X-Ray does add many classes to your distribution, do ensure you are using at least the 256MB AWS Lambda memory size.  This is explicitly set in manage.sh cmd_create(). Whilst the native image potentially can always use a lower memory setting, it would be recommended to keep the setting the same, especially to help compare performance.
+
+[[https]]
+== Using HTTPS or SSL/TLS
+
+If you code makes invokes HTTPS calls, such as to a micro-service (or AWS service), you will need to add configuration to the native image,
+as GraalVM will only include the dependencies when explictly declared.  Quarkus, by default enables this functionality on extensions that implicitly require it.
+For further information, please consult the link:native-and-ssl[Quarkus SSL guide]
+
+Open src/main/resources/application.properties and add the following line to enable SSL in your native image.
+
+[source]
+----
+quarkus.ssl.native=true
+----
+
+[[aws-sdk-v2]]
+== Using the AWS Java SDK v2
+
+With minimal integration, it is possible to leverage the AWS Java SDK v2,
+which can be used to invoke services such as SQS, SNS, S3 and DynamoDB.
+
+For native image, however the URL Connection client must be preferred over the ApacheHTTP Client
+when using synchronous mode, due to issues in the GraalVM compilation (at present).
+
+
+Create a `reflection-config.json` file in `src/main/resources/`
+[source, json]
+----
+[
+  {
+    "name":"com.sun.xml.internal.stream.XMLInputFactoryImpl",
+    "methods":[{"name":"<init>","parameterTypes":[] }]
+  }
+]
+----
```

Comment:
I think it would probably be better to use the `quarkus-jaxb` extension. I think it adds that class for reflection.

Reply:
Thanks for the tip.  I confirm that works in my sample project, updating the docs accordingly.